subprojects/messaging/src/main/java/org/gradle/internal/serialize/DefaultSerializerRegistry.java

```diff
@@ -206,6 +215,15 @@ private interface SerializerClassMatcherStrategy {
 
     }
 
+    /**
+     * Serializer wrapper, that allows specific instance to be created when they cannot be shared or reused.
+     *
+     * @param <S> The type supported by the serializer
+     */
+    public interface SerializerProvider<S> {
```

Comment:
`Provider` is overloaded here and can easily be confused with the provider api. Can we use a word like `Supplier` or `Factory` here?